### PR TITLE
Change mates2000.epd

### PR DESCRIPTION
**Pull request by vondele**
A random selection of those positions in `matetrace.epd` and `matedtrack.epd` that could be solved in less than 1M nodes by Stockfish.

More suitable for the purpose of this testset, namely a collection of faster to solve mates.

```
$ python matecheck.py --nodes 1000000 --epdFile mates2000.epd --engine ./stockfish.master
Loading FENs...
2000 FENs loaded...

Matetrack started for ./stockfish.master on mates2000.epd with --nodes 1000000 ...
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 134/134 [01:53<00:00,  1.19it/s]

Using ./stockfish.master on mates2000.epd with --nodes 1000000
Engine ID:     Stockfish dev-20240612-025da6a0
Total FENs:    2000
Found mates:   2000
Best mates:    2000
```

For stockfish CI (100k nodes), we have:
```
Using ./stockfish.master on mates2000.epd with --nodes 100000
Engine ID:     Stockfish dev-20240612-025da6a0
Total FENs:    2000
Found mates:   1220
Best mates:    1026
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -61,7 +61,7 @@ contains illegal moves or does not end in checkmate.
...
 * `matetrackpv.epd`: The same as `matetrack.epd`, but for each position the file also includes a PV leading to the checkmate, if such a PV is known.
 * `matedtrack.epd`: Derived from `matetrackpv.epd` by applying a best move in all those positions, where the winning side is to move, and where a best move is known. The order of the positions in `matedtrack.epd` corresponds 1:1 to the order in `matetrack.epd`. So the new test suite still contains 6556 mate problems, but for 6535 of them the side to move is going to get mated.
-* `mates2000.epd`: A smaller test suite with 2000 positions ranging from #1 to #7. In 1000 positions the side to move is going to get mated.
+* `mates2000.epd`: A smaller test suite with 2000 positions ranging from #1 to #27. A random selection of those positions in `matetrace.epd` and `matedtrack.epd` that could be solved in less than 1M nodes by Stockfish.
```
Comment (robertnurnberg) [inline README.md:64]: Typo matetrace.
Could say how many are to be mates?

**Comment by robertnurnberg** [comment]
Maybe we could sort the fens? Easily done with `sortmates.py --stable` for example.

**Comment by vondele** [comment]
> Maybe we could sort the fens? Easily done with `sortmates.py --stable` for example.

there is some advantage of keeping them unsorted, the batches created in matetrack will be more uniform and should run faster... (otherwise the large ones are possibly at the end).

**Comment by robertnurnberg** [comment]
> > Maybe we could sort the fens? Easily done with `sortmates.py --stable` for example.
> 
> there is some advantage of keeping them unsorted, the batches created in matetrack will be more uniform and should run faster... (otherwise the large ones are possibly at the end).

Could point. We could address this in a separate pr in the matecheck script itself? So that all users benefit. I.e. insert a shuffle before the chunking?

**Comment by vondele** [comment]
agreed, sorted.

**Comment by robertnurnberg** [comment]
For future reference:
```
Using ./sf17 on mates2000.epd with --nodes 1000000000
Engine ID:     Stockfish 17
Total FENs:    2000
Found mates:   1996
Best mates:    1994
```
So these mates are very likely optimal.

**Comment by robertnurnberg** [comment]
And here the results for a `10**10` test run with huntsman:

```
Using ../matetools/h1 on mates2000.epd with --nodes 10000000000
Engine ID:     The Huntsman 1
Total FENs:    2000
Found mates:   2000
Best mates:    1998
```

**Pull request merged**